app/post.go

```diff
@@ -28,14 +30,29 @@ func mdToHTML(md []byte) []byte {
 }
 
 func postHandler(c *gin.Context, app_settings common.AppSettings, database database.Database) ([]byte, error) {
+
+	// Get the post ID from the URL
+	postIDStr := c.Param("id")
+
+	// converting our postID string to an integer
+	postID, err := strconv.Atoi(postIDStr)
+
+	if err != nil || postID < 0 {
+		c.JSON(http.StatusBadRequest, gin.H{"error": "Invalid post ID"})
+
+		return nil, err
+	}
+
 	var post_binding common.PostIdBinding
```

Comment:
This should be at the top. This is the code that parses the uri param to the `post_binding.Id`.